skops/io/tests/test_persist.py

```diff
@@ -804,3 +804,75 @@ def test_loads_from_str():
     msg = "Can't load skops format from string, pass bytes"
     with pytest.raises(TypeError, match=msg):
         loads("this is a string")
+
+
+class _BoundMethodHolder:
+    """Used to test the ability to serialize and deserialize bound methods"""
+
+    def __init__(self, object_state: str):
+        self.object_state = (
+            object_state  # Initialize with some state to make sure state is persisted
+        )
+        self.chosen_function = (
+            np.log  # bind some method to this object, could be any persistable function
+        )
+
+    def bound_method(self, x):
+        return self.chosen_function(x)
+
+
+class TestPersistingBoundMethods:
+    @staticmethod
+    def assert_transformer_persisted_correctly(
+        loaded_transformer: FunctionTransformer,
+        original_transformer: FunctionTransformer,
+    ):
+        """Checks that a persisted and original transformer are equivalent, including the func passed to it
+        """
+        assert loaded_transformer.func.__name__ == original_transformer.func.__name__
+
+        assert_params_equal(
+            loaded_transformer.func.__self__.__dict__,
+            original_transformer.func.__self__.__dict__,
+        )
+        assert_params_equal(loaded_transformer.__dict__, original_transformer.__dict__)
+
+    @staticmethod
+    def assert_bound_method_holder_persisted_correctly(
+        original_obj: _BoundMethodHolder, loaded_obj: _BoundMethodHolder
+    ):
+        """Checks that the persisted and original instances of _BoundMethodHolder are equivalent
+        """
+        assert original_obj.bound_method.__name__ == loaded_obj.bound_method.__name__
+        assert original_obj.chosen_function == loaded_obj.chosen_function
+
+        assert_params_equal(original_obj.__dict__, loaded_obj.__dict__)
+
+    def test_for_base_case_returns_as_expected(self, tmp_path):
+        initial_state = "This is an arbitrary state"
+        obj = _BoundMethodHolder(object_state=initial_state)
+        bound_function = obj.bound_method
+        transformer = FunctionTransformer(func=bound_function)
+
+        loaded_transformer = save_load_round(transformer, tmp_path / "file.skops")
```

Comment:
Just FYI, we added `dumps` and `loads` now, so you can also do `loaded_transformer = loads(dumps(transformer))`. There will be a refactor to change this throughout all tests (#198) so it's okay to keep it like this.

Reply:
Oh nice! I might as well get ahead of it nonetheless 👍